### ISSUE TITLE
Testing of new data sets before public release

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/remote-datasets#9]
Pinging @GenericMappingTools/gmt-team. I have uploaded new remote data grids to the test server.  These are described [here](https://remote-datasets-my52fj7t6-gmt.vercel.app).  Note that until we complete the **dvc** move for test images we have not yet updated the _earth_relief_ files (they are still based off SRTM15+ v2.1 from 2019).  Once that is settled I will update those files to 2.3 and add the _synbath_ grid as well.  Because of this, if you run all the GMT tests they still all pass since all the DEM images are made using the same DEM as was used in the originals (once we update the DEM we will have ~60 failures...).

In the meantime, I hope you can find time to play a bit with the new datasets.  To do so, and not mess up your other remote files, I suggest you do this:

Edit your ConfigUserAdvanced.cmake file (perhaps after updating via the Template) and uncomment this line:

`# set (GMT_DATA_SERVER "test")`

then rebuild GMT. To preserve your existing cache and server files, simply set them aside via

```
mv ~/.gmt/server ~/.gmt/server_orig
mv ~/.gmt/cache ~/.gmt/cache_orig
```

so that you will only use files from the test server.  You can now try out making maps with `@earth_{gebco,gebcosi,geoid,mag,mag4km,vgg,wdmam}` - they all have their own default CPTs that will be downloaded via the cache.

Let me know if you find anything odd. The default CPTs are the ones recommended by the data provider, except Gebco which has not replied.  I am therefore using the same geo CPT for all the reliefs.

Once you have finished testing, remove the new directories and rename the old:

```
rm -f /.gmt/server  ~/.gmt/cache 
mv ~/.gmt/server_orig ~/.gmt/server 
mv ~/.gmt/cache_orig ~/.gmt/cache
```

**Comment by Esteban82** [comment]
@Paul I could only make maps from the GEBCO grids and look fine.

For the other grids (geoid,mag,mag4km,vgg,wdmam) I got this message: 
Are those grids available? Or am I doing something wrong? 


```
gmt grdimage @earth_mag -JW15c -Bg0 -Rd -png geoid


grdimage [ERROR]: Libcurl Error: HTTP response code said error
grdimage [WARNING]: You can turn remote file download off by setting GMT_DATA_UPDATE_INTERVAL to "off"
grdimage [ERROR]: File /home/federico/.gmt/cache/earth_mag was not found
grdimage [ERROR]: Cannot find file /home/federico/.gmt/cache/earth_mag
grdimage [ERROR]: File /home/federico/.gmt/cache/earth_mag was not found
grdimage [ERROR]: Cannot find file /home/federico/.gmt/cache/earth_mag
grdimage [ERROR]: File /home/federico/.gmt/cache/earth_mag not found
[Session gmt (0)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
```

**Comment by PaulWessel** [comment]
Interesting, it works if we specify a resolution:

`gmt grdimage @earth_mag_06m -JM12c -RFR -B -pdf t`

I will have a look.

**Comment by Esteban82** [comment]
I have an error with finer resolutions 

```

gmt grdimage @earth_gebco -JM15c -Bg0 -png gebco -R44/50/33/40
grdblend [NOTICE]: Remote data courtesy of GMT data server test [http://test.generic-mapping-tools.org]

grdblend [NOTICE]: GEBCO Earth Relief at 3x3 arc seconds tiles provided by SRTMGL3 (land only) [NASA/USGS].
grdblend [NOTICE]:   -> Download 1x1 degree grid tile (earth_gebco_03s_g): N33E044
grdblend [ERROR]: Libcurl Error: HTTP response code said error
grdconvert [ERROR]: File /home/federico/.gmt/server/earth/earth_gebco/earth_gebco_03s_g/N33E044.earth_gebco_03s_g.jp2 was not found
grdconvert [ERROR]: Cannot find file /home/federico/.gmt/server/earth/earth_gebco/earth_gebco_03s_g/N33E044.earth_gebco_03s_g.jp2
grdblend [ERROR]: ERROR - Unable to convert SRTM file /home/federico/.gmt/server/earth/earth_gebco/earth_gebco_03s_g/N33E044.earth_gebco_03s_g.jp2 to compressed netCDF format
grdblend [ERROR]: Unable to obtain remote file @N33E044.earth_gebco_03s_g.nc
grdblend [NOTICE]:   -> Download 1x1 degree grid tile (earth_gebco_03s_g): N33E044
grdblend [ERROR]: Libcurl Error: HTTP response code said error
...

```

**Comment by PaulWessel** [comment]
Thanks, this part is fixed via https://github.com/GenericMappingTools/gmt/pull/6151 but I also have to fix a few things with the server files.  Some tiles got labelled N30E300.earth_mag_02m_p.jp2 instead of N30W060.earth_mag_02m_p.jp2 for instance and thus could not be found.  But separate from the bug I fixed.  For example

`gmt grdimage @earth_faa -JM12c -RFR -B -pdf t`

now works but mag runs into missing filenames which I will fix next.

**Comment by PaulWessel** [comment]
OK, the mag files should be fine now.

**Comment by Esteban82** [comment]
In mag there is no data south of -60º. Is it a limitation of the original data?  
In the [docs](https://remote-datasets-git-new-dataset-additions-gmt.vercel.app/earth-mag.html) only says that: "This version is observed at sea level over oceanic regions and have no data over land."


`gmt grdimage @earth_mag_06m -JW15c -Bg0 -png mag -Rd`

![mag](https://user-images.githubusercontent.com/17908951/147397826-30ab2b6d-4b9f-4c1a-93d6-bf75c2daffc9.png)

**Comment by Esteban82** [comment]
Now I can download the rest of the grids.

I think that there is typo for the wdmam cpt. Grdimage searchs for earth_wdman.cpt (with N, not M).

```
gmt grdimage @earth_wdmam -JW15c -Bg0 -Rd -png wdmam

grdimage [ERROR]: GMT_Read_Data: File not found: earth_wdman.cpt
```

**Comment by PaulWessel** [comment]
I will check on the 60S issue.  The 1s and 3s SRTM tiles issue will take more time and is unresolved.

I have fixed the `wdman.cpt` typo. You will need to delete the gmt_data_server.txt file in your .gmt/server directory so it can be downloaded again with the fix.  Let me know if that works.

**Comment by PaulWessel** [comment]
The EMAG2 website only shows maps that stop at 60S and show nothing below 60S.  Seems like a cutoff.  The upward continued mag4km has data at the poles.

**Comment by PaulWessel** [comment]
The gebco issue should be fixed by https://github.com/GenericMappingTools/gmt/pull/6153.

**Comment by Esteban82** [comment]
Now, all the issues that I reported are working fine.

**Comment by Esteban82** [comment]
> The EMAG2 website only shows maps that stop at 60S and show nothing below 60S. Seems like a cutoff. The upward continued mag4km has data at the poles.

Yes, I check it. It seems that there is no official data below 60S.

**Comment by meghanrjones** [comment]
Why are there some redundant grid files at https://oceania.generic-mapping-tools.org/ and https://oceania.generic-mapping-tools.org/server/earth/earth_relief/ (e.g., earth_relief_01d.grd, earth_relief_60m.grd)?

**Comment by meghanrjones** [comment]
This may not be a cause for concern since it relates to non-updated files (`earth_relief_01d_g.grd`), but `test/api/apimat2grd.sh` fails when using the test server.

**Comment by meghanrjones** [comment]
I'm running into issues where the download is attempted using the `cache` dir rather than the appropriate `server` path:
```
(base) DarthMac:new-datasets meghanj$ gmt grdcut @earth_geoid_01d.grd -Rg -Ggeoid_01d.nc -V
grdcut [NOTICE]: Remote data courtesy of GMT data server test [http://test.generic-mapping-tools.org]

grdcut [NOTICE]: EGM2008 geoid at 1x1 arc degrees from Gaussian Cartesian filtering (111 km fullwidth) of the 1m grid [Pavlis et al., 2012].
grdcut [NOTICE]:   -> Download grid file [99K]: earth_geoid_01d_g.grd
grdcut [INFORMATION]: Downloading file http://test.generic-mapping-tools.org/cache/earth_geoid_01d.grd ...
grdcut [ERROR]: Libcurl Error: HTTP response code said error
```

**Comment by meghanrjones** [comment]
The `earth_geoid` colormap is missing: `grdimage [ERROR]: GMT_Read_Data: File not found: earth_geoid.cpt`.

**Comment by PaulWessel** [comment]
Yes it is.  I must have made an earth_geoid.cpt at home and forgot to add it.  I will do that later.  I know I have made geoid maps at home.

**Comment by PaulWessel** [comment]
Actually, no such thing.  I was using turbo.  So I have replaced @earth_geoid.cpt with turbo in the gmt_data_server.txt file and you will need to remove the old one or wait 24 hours..

**Comment by meghanrjones** [comment]
Running into another issue with the earth_geoid dataset:
```
gmt grdimage @earth_geoid_05m_g -Rg -JH15c -B -png test
```

```
grdblend [NOTICE]: EGM2008 geoid at 5x5 arc minutes from Gaussian Cartesian filtering (9 km fullwidth) of the 1m grid [Pavlis et al., 2012].
grdblend [NOTICE]:   -> Download 180x180 degree grid tile (earth_geoid_05m_g): S90W180
grdblend [ERROR]: Libcurl Error: HTTP response code said error
grdblend [WARNING]: You can turn remote file download off by setting GMT_DATA_UPDATE_INTERVAL to "off"
grdconvert [ERROR]: File /Users/meghanj/.gmt/server/earth/earth_geoid/earth_geoid_05m_g/S90W180.earth_geoid_05m_g.jp2 was not found
```

**Comment by PaulWessel** [comment]
That does it - I must have messed up something in the srv_tiler.sh script or similar.  I have noticed many tiles got named wrongly (some had E330 instead of W030 and now this with being named E180 but GMT looks for W180...  I may have to redo the processing once I find the culprit.

**Comment by meghanrjones** [comment]
OK. Another issue: I am confused why the following script tries to read a colormap from stdin rather than using the session cpt (this example works for the other datasets):

```
gmt begin test png
gmt grdimage @earth_mag_01d_g -Rg -JH15c -B
gmt colorbar -V
gmt end show
```

**Comment by PaulWessel** [comment]
All files for earth_geoid, earth_mag and earth_mag4km have been updated after fixing the -Rg vs -Rd issue.

**Comment by PaulWessel** [comment]
The cpt stdin issue seems to be a new bug (?) and affects older remote files as well.  Will be looking at that now.

**Comment by PaulWessel** [comment]
> OK. Another issue: I am confused why the following script tries to read a colormap from stdin rather than using the session cpt (this example works for the other datasets):
> 
> ```
> gmt begin test png
> gmt grdimage @earth_mag_01d_g -Rg -JH15c -B
> gmt colorbar -V
> gmt end show
> ```

This should be fixed by https://github.com/GenericMappingTools/gmt/pull/6171.

**Comment by meghanrjones** [comment]
The results for earth_mag_05m look off for both pixel and gridline:
![test_@earth_mag_05m_p](https://user-images.githubusercontent.com/14077947/148279875-39d2bdb8-0870-4be1-bd17-d8646dbe0a83.png)
Here's the same for earth_mag_06m:
![test_@earth_mag_06m_p](https://user-images.githubusercontent.com/14077947/148279920-9083988b-c012-41ff-ba71-43cb6b942c3c.png)

**Comment by PaulWessel** [comment]
Oddly the 6m grid has the specified scale of 0.2 and offset of 800 while the 5m grids have 0.03125 and 100, which what the values are for earth_vgg.  Not sure how that is possible but rerunning earth_mag now.

**Comment by PaulWessel** [comment]
Mystery solved and fixed: For the tiled resolutions we use JP2000 format on the server - which has no header etc.  When gmt downloads and converts these to netCDF locally it needs to know stuff - which it gets from gmt_data_server.txt.  If the author of that file forgets to set the metadata for the three mag data to, say, that of VGG then we notice once tiles are selected.  Also fixed gebco which has a scale of 1 since the authors of that data set made the decision to only store integer meters.

Solution for you is to wipe your gmt_server_data.txt file as well as the earth/* directory associated with mag, mag4km, wdmam, gebco, and gebcosi.

I made 05 and 06 maps for all these and all look fine now.

**Comment by PaulWessel** [comment]
I will make a test script for doing these 5 vs 6 maps for all data since a good and only check for this problem.

**Comment by meghanrjones** [comment]
> I will make a test script for doing these 5 vs 6 maps for all data since a good and only check for this problem.

Great, I think we could store useful test scripts like this in https://github.com/GenericMappingTools/remote-datasets/tree/main/scripts.

**Comment by meghanrjones** [comment]
A few questions/comments from running grdinfo on the new/updated datasets:
1. Why do some include the command (e.g., earth_faa) while others do not (e.g., earth_mag)?
2. For the gebco grids, "GEBCO Compilation Group, 2020 GEBCO 2021 Grid" should be "GEBCO Compilation Group, 2021 GEBCO 2021 Grid". Also, the DOI agrees with that written on their site, but still leads to a 404 error. Is there anything we can do here?
3. Most have a remark that makes it clear how the grid relates to the data source, except earth_faa and earth_vgg (e.g., `Obtained by Gaussian Cartesian filtering (111.2 km fullwidth) from /tmp/19294/extend.grd`. Should `from tmp...` be more descriptive?
4. The earth_synbath and earth_relief preprint citation is 2021 (https://doi.org/10.1002/essoar.10508279.2). Does the remark list 2022 instead because the publish date will change for the final version? If so will the DOI still be the same for the published version and preprint?
5. The year for earth_vgg citation is 2021 but the remark lists 2019 (https://doi.org/10.1016/j.asr.2019.09.011).

**Comment by meghanrjones** [comment]
For the gridline registered grids, there is often empty space between the map border and grid edge:
```
gmt grdimage @earth_synbath_01d_g -R110/140/20/35 -JB125/20/25/45/12c -B -png test_synbath_g
```
![test_synbath_g](https://user-images.githubusercontent.com/14077947/149202966-aa0de428-d21b-4daf-b15d-c561ab375c63.png)

**Comment by PaulWessel** [comment]
> For the gridline registered grids, there is often empty space between the map border and grid edge:
> 
> ```
> gmt grdimage @earth_synbath_01d_g -R110/140/20/35 -JB125/20/25/45/12c -B -png test_synbath_g
> ```
> 
> ![test_synbath_g](https://user-images.githubusercontent.com/14077947/149202966-aa0de428-d21b-4daf-b15d-c561ab375c63.png)

Probably related to recent change of the double-padding removal.  I will try to debug this and see why this case gets short-changed.

**Comment by PaulWessel** [comment]
Since this is a GMT bug, maybe create an issue on the GMT issues site using earth_relief instead?

**Comment by PaulWessel** [comment]
I will rebuild VGG, FAA but wait on Gebco.  I will see if I can learn why mag has no command.

**Comment by meghanrjones** [comment]
The PyGMT tests that use the SRTM data are failing with the test server due to tiles not being found. Can you try this to see if it works for you?
```
gmt grdcut @srtm_relief_03s_g -R135/136/35/36 -Gtest.nc
```

**Comment by PaulWessel** [comment]
I forgot to copy over the STRM tiles after I deleted and rebuilt the new earth_relief files... Now fixed.

**Comment by PaulWessel** [comment]
> Any plans to include the _International Bathymetric Chart of the Southern Ocean_ (IBCSO) and _International Bathymetric Chart of the Arctic Ocean_ (IBCAO)?
> Or are these included in one of the other data sets?

I thought they are in the GEBCO grid but perhaps not?  Anyway, they are not global grids.

**Comment by anbj** [comment]
> > Any plans to include the _International Bathymetric Chart of the Southern Ocean_ (IBCSO) and _International Bathymetric Chart of the Arctic Ocean_ (IBCAO)?
> > Or are these included in one of the other data sets?
> 
> I thought they are in the GEBCO grid but perhaps not? Anyway, they are not global grids.

I think you are right.

**Comment by PaulWessel** [comment]
FAA and VGG files updated on test server

**Comment by meghanrjones** [comment]
@GenericMappingTools/pygmt-maintainers @GenericMappingTools/gmt-maintainers, the plan is to update `earth_relief` and add the new datasets to the server on Jan 27, 2022.

**Comment by Esteban82** [comment]
Is there any problem with WDMAM data set? The codes works with other datasets (vgg, faa).

```
mt grdimage @earth_wdmam_02m -Rd -png mpa
grdimage [ERROR]: Libcurl Error: HTTP response code said error
grdimage [WARNING]: You can turn remote file download off by setting GMT_DATA_UPDATE_INTERVAL to "off"
grdimage [ERROR]: File /home/federico/.gmt/cache/earth_wdmam_02m was not found
grdimage [ERROR]: Cannot find file /home/federico/.gmt/cache/earth_wdmam_02m
grdimage [ERROR]: File /home/federico/.gmt/cache/earth_wdmam_02m was not found
grdimage [ERROR]: Cannot find file /home/federico/.gmt/cache/earth_wdmam_02m
grdimage [ERROR]: File /home/federico/.gmt/cache/earth_wdmam_02m not found
[Session gmt (0)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
```

**Comment by Esteban82** [comment]
Is it possible that the `02m` grid doesn't exist? And that the original version is `03m`?

**Comment by PaulWessel** [comment]
Correct, there is no such thing as 02m.  I see our docs is in error - will fix.
